examples_tests/61.BlitFilterTest/default_compute_normalization.comp

```diff
@@ -0,0 +1,192 @@
+#version 460 core
+
+#include <nbl/builtin/glsl/macros.glsl>
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_X_
+#define _NBL_GLSL_WORKGROUP_SIZE_X_ 8
+#endif
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_Y_
+#define _NBL_GLSL_WORKGROUP_SIZE_Y_ 8
+#endif
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_Z_
+#define _NBL_GLSL_WORKGROUP_SIZE_Z_ 8
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_BIN_COUNT_
+#define _NBL_GLSL_BLIT_NORMALIZATION_BIN_COUNT_ 256
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_
+#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ must be defined
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_
+#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_ must be defined
+#endif
+
+// #define _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+
+#if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage1D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image1D
+	#endif
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage2D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image2D
+	#endif
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage3D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image3D
+	#endif
```

Comment:
I think you could pick `imageND` first as `_NBL_GLSL_BLIT_NORMALIZATION_OUT_IMAGE_TYPE_IMPL_ and then later do

```c
#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
    #define _NBL_GLSL_BLIT_NORMALIZATION_OUT_IMAGE_TYPE_ u##_NBL_GLSL_BLIT_NORMALIZATION_OUT_IMAGE_TYPE_IMPL_
#else
    #define _NBL_GLSL_BLIT_NORMALIZATION_OUT_IMAGE_TYPE_ _NBL_GLSL_BLIT_NORMALIZATION_OUT_IMAGE_TYPE_IMPL_
#endif
```

Reply:
Resolved in the following commit.

Reply:
Ahh apologies, this slipped through an untested path.

This will not work:
```C
 #define _NBL_GLSL_BLIT_NORMALIZATION_OUT_IMAGE_TYPE_ u##_NBL_GLSL_BLIT_NORMALIZATION_OUT_IMAGE_TYPE_IMPL_
```

I would have to do:
```C
#define _NBL_GLSL_BLIT_OUT_IMAGE_TYPE_ NBL_GLSL_CONCATENATE2(u, _NBL_GLSL_BLIT_OUT_IMAGE_TYPE_IMPL_)
```

Reply:
yep do that.

Reply:
Resolved.